Test harness tool does not wait for CRDs to be processed

**What happened**:

https://circleci.com/gh/kudobuilder/kudo/9541 flaked with:
```
=== RUN   kudo
2020/01/14 14:24:53 CustomResourceDefinition:/instances.kudo.dev created
2020/01/14 14:24:53 CustomResourceDefinition:/operators.kudo.dev created
2020/01/14 14:24:53 CustomResourceDefinition:/operatorversions.kudo.dev created
--- FAIL: kudo (6.48s)
    harness.go:215: Running tests with a mocked control plane (kube-apiserver and etcd).
    harness.go:196: started test environment (kube-apiserver and etcd) in 6.345799792s
    harness.go:412: error creating resource Operator:default/configmap-operator: no matches for kind "Operator" in version "kudo.dev/v1beta1"
    harness.go:461: tearing down mock control plane
FAIL
```

A retry succeeded.

**What you expected to happen**:

No failure! The tool should wait for all CRDs to reach usable `status` before trying to create objects of the newly defined types.

**How to reproduce it (as minimally and precisely as possible)**:

`go run ./cmd/kubectl-kudo test` many times

--- a/pkg/test/utils/kubernetes.go
+++ b/pkg/test/utils/kubernetes.go
@@ -789,19 +789,25 @@ func WaitForCRDs(dClient discovery.DiscoveryInterface, crds []runtime.Object) er
 
 	for _, crdObj := range crds {
 		if !MatchesKind(crdObj, crdKind) {
-			continue
+			return fmt.Errorf("the following passed object does not match %v: %v", crdKind, crdObj)
 		}
 
-		crd, ok := crdObj.(*apiextensions.CustomResourceDefinition)
-		if !ok {
-			continue
+		switch crd := crdObj.(type) {
+		case *apiextensions.CustomResourceDefinition:
+			waitingFor = append(waitingFor, schema.GroupVersionKind{
+				Group:   crd.Spec.Group,
+				Version: crd.Spec.Version,
+				Kind:    crd.Spec.Names.Kind,
+			})
+		case *unstructured.Unstructured:
+			waitingFor = append(waitingFor, schema.GroupVersionKind{
+				Group:   crd.Object["spec"].(map[string]interface{})["group"].(string),
+				Version: crd.Object["spec"].(map[string]interface{})["version"].(string),
+				Kind:    crd.Object["spec"].(map[string]interface{})["names"].(map[string]interface{})["kind"].(string),
+			})
+		default:
+			return fmt.Errorf("the folllowing passed object is not a CRD: %v", crdObj)
 		}
-
-		waitingFor = append(waitingFor, schema.GroupVersionKind{
-			Group:   crd.Spec.Group,
-			Version: crd.Spec.Version,
-			Kind:    crd.Spec.Names.Kind,
-		})
 	}
 
 	return wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (done bool, err error) {

